bin/combine_meta.py

```diff
@@ -0,0 +1,258 @@
+        )
+        logging.warning(warning_message)
+        print_samplename = False
+    return print_samplename
+
+
+def validate_readcounts(
+    sample: str, meta_list: List[Dict[str, Any]], print_samplename: bool
+) -> bool:
+    """
+    Check if there IRODS total_counts equals samtools output
+    sample (str): sample name
+    meta_list (List[Dict[str, Any]]): a list containing metadata for all files of a particular sample
+    print_samplename: (bool): whether we should print a header with sample name
+    """
+    # get samples with inconsistent total number of reads
+    warning_list = [
+        cram_meta["cram_path"]
+        for cram_meta in meta_list
+        if cram_meta["total_reads"] != cram_meta["num_reads_processed"]
+    ]
+    if warning_list:
+        # print a header
+        raise_sample_warning(sample, print_samplename)
+        # make a warning message
+        warning_message = (
+            "IRODS total_count != num_reads_processed for files:"
+            + ",".join(warning_list)
+        )
+        logging.warning(warning_message)
+        print_samplename = False
+    return print_samplename
+
+
+def validate_atac(
+    sample: str, meta_list: List[Dict[str, Any]], print_samplename: bool
+) -> bool:
+    """
+    Check if there IRODS total_counts equals samtools output
+    sample (str): sample name
+    meta_list (List[Dict[str, Any]]): a list containing metadata for all files of a particular sample
+    print_samplename: (bool): whether we should print a header with sample name
+    """
+    warning_list = [
+        cram_meta["cram_path"]
+        for cram_meta in meta_list
+        if "atac" in cram_meta["library_type"].lower() and cram_meta["i2len"] == "24"
+    ]
+    if warning_list:
+        # print a header
+        raise_sample_warning(sample, print_samplename)
+        # make a warning message
+        warning_title = f"The following files are suspected to be 10X ATAC. They were renamed according to CellRanger naming convention :"
+        warning_message = warning_title + ",".join(warning_list)
+        logging.warning(warning_message)
+        print_samplename = False
+    return print_samplename
+
+
+def validate_metalist(meta_list: List[Dict[str, Any]], validate_all: bool) -> None:
+    """
+    Validates metadata values in a list of columns
+    meta_list (List[Dict[str, Any]]): a list containing metadata for all files of a particular sample
+    warning_messages (List[str]): a list of warning for particular sample
+    """
+    # get sample indexes
+    sample_index = get_sampleindex(meta_list)
+
+    # check if there are dulicated filenames
+    validate_filenames(meta_list)
+
+    # validate cram files for each sample
+    for sample, indexes in sample_index.items():
+        print_samplename = True
+        # subsample metadata list
+        subsample_metalist = [meta_list[idx] for idx in indexes]
+        # validate metadata
+        print_samplename = validate_consistency(
```

Comment:
The reuse of `print_samplename`, particularly after `validate_all`, is odd. 
Could you add a comment as what's happening there?